### ISSUE TITLE
[BFW-6830][BUG] Loaded XL carriage crashes into parked tool

### ISSUE THREAD

**Issue by muesli** [prusa3d/Prusa-Firmware-Buddy#4259]
### Printer model

XL

### Firmware version

6.2.0-alpha2

### Upgrades and modifications

None

### Printing from...

PrusaLink

### Describe the bug

I cancelled a print after it had done its first layer, because I wasn't happy with the bed adhesion. The carriage (loaded with Tool 1) stopped at the back/right, just in front of Tool 5, when the print got cancelled.

I then selected "Park Tool" from the menu, which caused the carriage to ram tool 1 into tool 5 before recovering and parking the loaded tool 1.

### How to reproduce

I don't dare attempting to reproduce it, it looks like my tool 5 got damaged by this.

### Expected behavior

Never ram into another tool. The printer must have known the current X/Y position of the carriage and should prevent this.

**Comment by muesli** [comment]
Sounds similar/related to #4154.

**Comment by KamilKeski** [comment]
I experienced the same on the first print I tried with this version.  Tried rehoming to no avail.  It keeps crashing into tool 5.  I gave up on the second attempt as it's so violent I fear I've already caused damage.

**Comment by CZDanol** [comment]
Hey, thanks for the reports. We have identified some motion-related issues in our 6.2.0 and they should hopefully be fixed in the next candidate firmware.

**Comment by muesli** [comment]
Gotta hate that GitHub Action. Yes, that ticket is still valid otherwise I would have closed it myself.

**Comment by CZDanol** [comment]
So this is still happening on 6.2.0 final?

**Comment by KamilKeski** [comment]
I have not experienced this issue on 6.2.0.  I've been through many tool changes, rehomings, etc on the final release without this particular issue reoccurring.  From my personal experience this was easy to reproduce with pre-release and I cannot reproduce with release.

My $0.02 - I consider my issue resolved.

**Comment by CptDelay** [comment]
For me it started to happen since the upgrade to 6.2.1. At least thats when i observed it for the first time. It seems to happen quite reproducible too. Whenever i cancel a Multi-Material Print when Toolhead 2 is active (2 Toolhead Version of the XL), it parks in front of Toolhead 1. Seems to be the same Spot where it parks after a succesful print, except it doesn't park the Tool first. When i command it to park the active Tool via the Control Menu it drives the Toolhead back into Toolhead 1 while starting its homing sequence. Only workaround is to manually release the Tool and dock it by hand.

**Comment by dadinthegarage** [comment]
I am having the same issue with 6.2.0.

**Comment by dadinthegarage** [comment]
I just had it happen again. Cancelled a print, 6.2.2, 5 TH. It was in the probing points mode, I cancelled print, it went back and crashed into tool 1. I did not get a crash error. Almost seemed like it went too far back. I then went to control and park current tool, went fine.

**Comment by schrodlm** [comment]
I'll have a look

**Comment by CZDanol** [comment]
FIY @schrodlm  there is a bug in aborting prints during the MBL that is fixed in the private. Could have been caused by that one.

**Comment by alexandru-g** [comment]
I also have issues with the carriage crashing into parked tools. I've seen it in 6.2.4 for sure (just updated to 6.2.5 but I don't see it fixing the issue). The way it replicates (consistently) for me is this:

1. Abort print
2. Carriage moves to printer's back right (near tool 1)
3. Park current tool from menu. The printer will try and home, but something fails and it starts backing up into tool 1.

Randomly, after finishing a print, the next home is not performed "as planned". X or Y hits the gantry hard and the motor loudly skips and then it starts to move with those "unsure steps" on the X axis until it detects the axis start. Lets say this isn't such a big issue, but this morning it started doing thos "unsure steps" on the Y axis towards tool 1 again.



I don't believe there is something wrong hardware-wise with my machine, it works perfectly good in 99.9% of the time.

**Comment by Ingurum** [comment]
I have the same issue 5TH, whenever i cancel a print the toolhead moves in front of tool1. Whenever something happens next (new print, park tool or whatever) the toolhead backs into tool1. I have to manually move the head to Y0 and do auto home.

**Comment by Gik-0101** [comment]
I can also confirm this error.

FW: 6.2.5

This error has been around for a while now, and (if careless) it can seriously damage the printer.

Is there any information about a release of a new firmware for the XL that fixes this error?

It's great if other printers are improved too, but I don't own them, and the XL is still suffering.
So my focus is purely on the XL.

Please don't let the XL fall into oblivion.

**Comment by ssill2** [comment]
I can confirm I see this in 6.2.6 too.  I've not tried in 6.4.0 yet.  I might this weekend.  I saw other issues and and reverted back to 6.2.6 thinking it was a safer firmware.  I don't typically have to cancel prints often, but I was doing a tpu print with five colors and I tend to have to cancel prints to tune settings and reprint so this was a real bummer.

**Comment by alexandru-g** [comment]
I have been using 6.4.0-RCx and now 6.4.0 stable for a while and I can confirm the bug is gone for me.

**Comment by ssill2** [comment]
Thanks I may give it a go this evening then.

**Comment by crasbe** [comment]
This is not fixed yet with 6.4.0, see #4662 and the linked issues there.

**Comment by muesli** [comment]
Can also confirm this is still happening with 6.4.0.

**Comment by crasbe** [comment]
It would be nice to get feedback if the internal ticket is completed before having this closed 🤔

**Comment by ssill2** [comment]
yeah I haven't had time to do a ton of testing on this.  But I can confirm that I do get crash when cancelling a print job that was multi-material.  with latest firmware.  Not that I'm sure it matters, but the print was with TPU and PLA as supports

**Comment by crasbe** [comment]
It would be nice to get feedback if the internal ticket is completed before having this closed 🤔
